src/ArrayToXml.php

```diff
@@ -99,6 +107,22 @@ public function toDom()
         return $this->document;
     }
 
+    /**
+     * Sets dom properties on $this->document.
+     *
+     * @param $domProperties
+     * @throws \Exception
+     */
+    public function setDomProperties($domProperties)
+    {
+        foreach ($domProperties as $key => $value) {
```

Comment:
Extract this to a new `protected` function called `ensureValidDomProperties`

Reply:
@freekmurze i would rather prefer to use something like "isValidDomProperty" and just extract the property_exists part to it. Otherwise we need to traverse the property array twice. Is that ok for you?.

Reply:
I don't mind double traversion.

Reply:
@freekmurze done as well